test/compatibility/compatibility_test.py

```diff
@@ -0,0 +1,170 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import inspect, os, re, shutil, sys, unittest
+
+class CompatibilityTestCase(unittest.TestCase):
+    '''
+    The base class for running compatibility tests.
+    '''
+
+    def pr(self, s):
+        '''
+        Print diagnostic output.
+        '''
+        print(s)
+
+    def branch_path(self, branch):
+        '''
+        Get path to a branch.
+        '''
+        return os.path.abspath(os.path.join('..', branch))
+
+    def module_file(self):
+        '''
+        Get the (base) file name for this test.
+        '''
+        return os.path.basename(inspect.getfile(self.__class__))
+    
+    def module_name(self):
+        '''
+        Get the module name for this test.
+        '''
+        return self.module_file().replace('.py', '')
+
+    def system(self, command):
+        '''
+        Run a command, fail the test if the command execution failed.
+        '''
+        self.assertEqual(os.system(command), 0)
+
+    def prepare_wt_branch(self, branch):
+        '''
+        Check out and build another WT branch.
+        '''
+
+        # Clone the repository and check out the correct branch
+        path = self.branch_path(branch)
+        if os.path.exists(path):
+            self.pr(f'Branch {branch} is already cloned')
+            self.system(f'git -C "{path}" pull')
+        else:
+            source = 'git@github.com:wiredtiger/wiredtiger.git'
+            self.pr(f'Cloning branch {branch}')
+            self.system(f'git clone "{source}" "{path}" -b {branch}')
+
+        # Build
+        build_path = os.path.join(path, 'build')
+        if not os.path.exists(os.path.join(build_path, 'build.ninja')):
+            os.mkdir(build_path)
+            # Disable WT_STANDALONE_BUILD, because it is not compatible with branches 6.0 and
+            # earlier.
+            cmake_args = '-DWT_STANDALONE_BUILD=0 -DENABLE_PYTHON=1'
+            self.system(f'cd "{build_path}" && cmake {cmake_args} -G Ninja ../.')
```

Comment:
This won't work on 5.0 and previous branches. It is fine to leave it like this if they won't have this test.

Reply:
Sounds good. Let me add a comment explaining that, so that we would not be caught by surprise when we decide to use this for the 5.x or 4.4 branches.